CLAUDE.md

```
@@ -1,270 +1,1169 @@
+
+- **No deprecation annotations**: Do not use `#[deprecated]`, `#[allow(dead_code)]`, or similar annotations to preserve unused code
+- **Binary-only interface**: This project does not maintain API compatibility for external consumers - all code must serve the current CLI functionality
+- **Dead code elimination**: Aggressively remove any unused functions, methods, structs, or modules during refactoring
+- **Immediate cleanup**: When refactoring or implementing features, remove unused code paths immediately rather than marking them for future removal
+
+#### Documentation Research Hierarchy
+
+When implementing or researching functionality, follow this order:
+
+1. **FIRST**: Generate and examine local documentation
+   ```bash
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should be surrounded by blank lines
null

(MD031, blanks-around-fences)